model/src/main/java/org/cloudfoundry/identity/uaa/provider/PasswordPolicy.java

```diff
@@ -16,22 +16,18 @@
 
 
 import java.util.Date;
+import org.cloudfoundry.identity.uaa.authentication.GenericPasswordPolicy;
 
-public class PasswordPolicy {
+public class PasswordPolicy extends GenericPasswordPolicy<PasswordPolicy> {
 
     public static final String PASSWORD_POLICY_FIELD = "passwordPolicy";
 
-    private int minLength;
-    private int maxLength;
-    private int requireUpperCaseCharacter;
-    private int requireLowerCaseCharacter;
-    private int requireDigit;
-    private int requireSpecialCharacter;
     private int expirePasswordInMonths;
     private Date passwordNewerThan;
 
     public PasswordPolicy() {
-        minLength = maxLength = requireUpperCaseCharacter = requireLowerCaseCharacter = requireDigit = requireSpecialCharacter = expirePasswordInMonths = -1;
+        super();
+        this.expirePasswordInMonths = -1;
```

Comment:
We should probably rename expirePasswordInMonths to expireInMonths and move it to the GenericPasswordPolicy.